File: Iniciante/1047/1047.c
#include <stdio.h>
 
int main() {
    
    int h1, m1, h2, m2, th, tm;
    scanf("%d%d%d%d", &h1, &m1, &h2, &m2);
    
    if ((m1 == m2) && (h1 == h2)) {
        tm = m2 - m1;
        th = 24;
    }
    
    else {   
        if ((m2 - m1) >= 0)
            tm = m2 - m1;
        else {
            m2 = m2 + 60;
            tm = m2 - m1;
            h2--;
        }
        
        if ((h2 - h1) >= 0)
            th = h2 - h1;
        else {
            h2 = h2 + 24;
            th = h2 - h1;
        }
    }    
    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", th, tm);
    return 0;
}

File: Python/Iniciante/1047.py
# Problema 1047 - Beecrowd - Iniciante - Nível 9

# Inicalmente, vamos ler os valores e salvar em variáveis
horaInicial, minutoInicial, horaFinal, minutoFinal = map(int, input().split(' '))

# Caso as horas de inicio e final sejam iguais
if horaInicial == horaFinal:

    # Caso os minutos de inicio e final sejam iguais
    if minutoInicial == minutoFinal:
        print(f'O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)')
    
    # Caso os minutos de inicio sejam maiores que o final
    elif minutoInicial > minutoFinal:
        print(f'O JOGO DUROU 23 HORA(S) E {60 - (minutoInicial - minutoFinal)} MINUTO(S)')

    # Caso os minutos de inicio sejam menores que o final
    elif minutoInicial < minutoFinal:
        print(f'O JOGO DUROU 0 HORA(S) E {minutoFinal - minutoInicial} MINUTO(S)')

# Caso a hora de inicio seja maior que a final
elif horaInicial > horaFinal:

    # Caso os minutos de inicio e final sejam iguais
    if minutoInicial == minutoFinal:
        print(f'O JOGO DUROU {24 - (horaInicial - horaFinal)} HORA(S) E 0 MINUTO(S)')
    
    # Caso os minutos de inicio sejam maiores que o final
    elif minutoInicial > minutoFinal:
        print(f'O JOGO DUROU {23 - (horaInicial - horaFinal)} HORA(S) E {60 - (minutoInicial - minutoFinal)} MINUTO(S)')

    # Caso os minutos de inicio sejam menores que o final
    elif minutoInicial < minutoFinal:
        print(f'O JOGO DUROU {24 - (horaInicial - horaFinal)} HORA(S) E {minutoFinal - minutoInicial} MINUTO(S)')

# Caso a hora de inicio seja menor que a final
elif horaInicial < horaFinal:

    # Caso os minutos de inicio e final sejam iguais
    if minutoInicial == minutoFinal:
        print(f'O JOGO DUROU {horaFinal - horaInicial} HORA(S) E 0 MINUTO(S)')
    
    # Caso os minutos de inicio sejam maiores que o final
    elif minutoInicial > minutoFinal:
        print(f'O JOGO DUROU {horaFinal - horaInicial - 1} HORA(S) E {60 - (minutoInicial - minutoFinal)} MINUTO(S)')

    # Caso os minutos de inicio sejam menores que o final
    elif minutoInicial < minutoFinal:
        print(f'O JOGO DUROU {horaFinal - horaInicial} HORA(S) E {minutoFinal - minutoInicial} MINUTO(S)')